debezium-core/src/main/java/io/debezium/heartbeat/Heartbeat.java

```diff
@@ -27,6 +27,11 @@
  */
 public interface Heartbeat {
 
+    @FunctionalInterface
```

Comment:
This comes back to my question about `onOplogEvent()` above. Is the new method actually needed?

Reply:
@gunnarmorling To obtain new offsets for MongoDB one has to parse a JSON object which I consider to be expensive operation. Hence I extended API to extract offsets only when heartbeat is necessary. Otherwise we would need to intorudce a method like `willBeat` returning true if the heartbeat is going to be emitted in next call and execute parsing only in such case.